driver/src/main/java/org/neo4j/driver/v1/Result.java

```diff
@@ -50,49 +66,40 @@
     boolean next();
 
     /**
-     * From the current record the result is pointing to, retrieve the value in the specified field.
+     * Advance the cursor as if calling next multiple times
      *
-     * @param fieldIndex the field index into the current record
-     * @return the value in the specified field
+     * @throws IllegalArgumentException if records is negative
+     * @return the actual number of records successfully skipped
      */
-    Value get( int fieldIndex );
+    int skip( int records );
 
     /**
-     * From the current record the result is pointing to, retrieve the value in the specified field.
-     * If no value could be found in the specified filed, null will be returned.
+     * Move to the first record if possible, otherwise do nothing
      *
-     * @param fieldName the field to retrieve the value from
-     * @return the value in the specified field or null if no value could be found in the specified filed
+     * @return true if the cursor was successfully placed at the first record
      */
-    Value get( String fieldName );
+    boolean first();
 
     /**
-     * Get an ordered sequence of the field names in this result.
+     * Move to the first record if possible and verify that it is the only record
      *
-     * @return field names
+     * @return true if the cursor was successfully placed at the single first and only record
      */
-    Iterable<String> fieldNames();
+    boolean single();
 
     /**
-     * Retrieve the first field of the next record in the stream, and close the stream.
-     *
-     * This is a utility for the common case of statements that are expected to yield a single output value.
-     *
-     * <pre>
-     * {@code
-     * Record record = statement.run( "MATCH (n:User {uid:..}) RETURN n.name" ).single();
-     * }
-     * </pre>
+     * Retrieve and store the entire remaining result stream (including the current record).
+     * This can be used if you want to iterate over the stream multiple times or to store the
+     * whole result for later use.
      *
-     * @return a single record from the stream
-     * @throws ClientException if the stream is empty
+     * Calling this method exhausts the result cursor and moves it to the last record
      */
-    Record single();
+    List<Record> retain();
```

Comment:
because I felt it is confusing that asList in one case returns a list of values but a list of records.